router-tests/router_plugin_test.go

```diff
@@ -0,0 +1,36 @@
+package integration
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/wundergraph/cosmo/router-tests/testenv"
+	"github.com/wundergraph/cosmo/router/core"
+	"github.com/wundergraph/cosmo/router/pkg/config"
+)
+
+func TestRouterPlugin(t *testing.T) {
+	t.Parallel()
+
+	testenv.Run(t, &testenv.Config{
+		RouterConfigJSONTemplate: testenv.ConfigWithPluginsJSONTemplate,
+		RouterOptions: []core.Option{
+			core.WithPlugins(config.PluginsConfiguration{
+				Path: "../router/plugins",
```

Comment:
Perhaps point this at something in `../demo` or testdata or something, I don't like this being dependent on the current state of the router dir.